system-addon/content-src/components/PreferencesPane/PreferencesPane.jsx

```diff
@@ -79,13 +85,23 @@ class PreferencesPane extends React.Component {
               <PreferencesInput className="showTopSites" prefName="showTopSites" value={prefs.showTopSites} onChange={this.handlePrefChange}
                 titleString={{id: "settings_pane_topsites_header"}} descString={{id: "settings_pane_topsites_body"}} />
 
+              <div className="options">
+                <PreferencesInput className="showMoreTopSites" prefName="topSitesCount" value={prefs.topSitesCount !== TOP_SITES_DEFAULT_LENGTH} onChange={this.handlePrefChange}
+                  titleString={{id: "settings_pane_topsites_options_showmore"}} labelClassName="icon icon-topsites" />
+              </div>
+
               {sections
                 .filter(section => !section.shouldHidePref)
                 .map(({id, title, enabled, pref}) =>
                   <PreferencesInput key={id} className="showSection" prefName={(pref && pref.feed) || id}
                     value={enabled} onChange={(pref && pref.feed) ? this.handlePrefChange : this.handleSectionChange}
                     titleString={(pref && pref.titleString) || title} descString={pref && pref.descString} />)}
 
+              {this.topStoriesOptions && !this.topStoriesOptions.hidden &&
+                <PreferencesInput className="showTopStories" prefName="feeds.section.topstories"
+                  value={prefs["feeds.section.topstories"]} onChange={this.handleChange}
+                  titleStringId="header_recommended_by" titleStringValues={{provider: this.topStoriesOptions.provider_name}}
+                  descStringId={this.topStoriesOptions.provider_description} />}
```

Comment:
Why was this stories-specific prefs added? It doesn't seem to do anything? And it's already handled by the sections code… ?

Reply:
In particular `this.topStoriesOptions` is `undefined`.

Reply:
oy. That was a screwup when fixing merge conflicts. Lemme fix real quicl